contracts/V3Router.sol

```diff
@@ -94,15 +94,17 @@ contract V3Router is BaseStrategyInitializable {
             }
         }
 
-        (uint256 _amountFreed, ) = liquidatePosition(
-            _debtOutstanding + _profit
-        );
-
-        _debtPayment = Math.min(_debtOutstanding, _amountFreed);
+        uint256 _amountFreed;
+        (_amountFreed, _loss) = liquidatePosition(_debtOutstanding + _profit);
 
-        //Adjust profit in case we had any losses from liquidatePosition
-        unchecked {
-            _profit = _amountFreed - _debtPayment;
+        if (_loss > _profit) {
+            _profit = 0;
+            _loss -= _profit;
+            _debtPayment = _amountFreed;
+        } else {
+            _loss = 0;
+            _profit -= _loss;
+            _debtPayment = _debtOutstanding;
```

Comment:
This might not work in some cases. For example:

Assume strategy has:
100 debt
90 ta
and 50 debtOutstanding

you will first calculate the profit as 0 and loss as 10.

Then, we will withdraw 50 assets and let's say we done it without a loss, meaning that we exactly withdrawn 50 assets. So `liquidatePosition` returned loss value is 0 which overriden our previously calculated `_loss`, which was 10

Then basically we didn't report any loss or profit although we had to report a loss.

Reply:
I think you need to first get the general loss via debt/totalassets method. Then, consider the withdrawal losses (from liquidatePosition)

Reply:
fixed https://github.com/Schlagonia/v3-router/pull/3/commits/a1ae210364c84db5f89dfcf89a3e705cc5fc96e8